### PR TITLE
pokemon

### PR DESCRIPTION
**Pull request by jiangzhengwenjz**
* Decompiled pokemon_3.s into pokemon.c
* Cleaned up pokemon.h
* Related data decomp

### CODE REVIEW

File: include/field_specials.h [hunk 1]
```diff
@@ -11,5 +11,7 @@ void sub_80CCB68(void);
 void nullsub_60(void);
 void nullsub_61(void);
 u8 ContextNpcGetTextColor(void);
+void set_unknown_box_id(u8);
```
Comment (garakmon) [inline include/field_specials.h]: caps?

File: include/party_menu.h [hunk 2]
```diff
@@ -61,5 +61,6 @@ void sub_8126894(u8 taskId, TaskFunc followUpFunc);
 void sub_81279E0(void);
 void ItemUseCB_Medicine(u8 taskId, TaskFunc followUpFunc);
 u8 GetItemEffectType(u16 itemId);
+u8 pokemon_order_func(u8);
```
Comment (garakmon) [inline include/party_menu.h]: caps?

File: src/pokemon.c [hunk 3]
```diff
@@ -5604,8 +7137,728 @@ bool8 IsShinyOtIdPersonality(u32 otId, u32 personality)
...
+        structPtr->templates[i] = sOakSpeechNidoranFDummyTemplate;
+        for (j = 0; j < structPtr->frameCount; ++j)
+            structPtr->frameImages[i * structPtr->spriteCount + j].data = &structPtr->bufferPtrs[i][j * 0x800];
+        structPtr->templates[i].images = &structPtr->frameImages[i * structPtr->spriteCount]; // should be frameCount logically
+        structPtr->templates[i].anims = gUnknown_82349BC;
+        structPtr->templates[i].paletteTag = i;
+    }
+}
+
+#ifdef NONMATCHING
```
Comment (garakmon) [inline src/pokemon.c]: note to self: take a look at this

File: include/field_specials.h (hunk 1 above)
Comment (jiangzhengwenjz) [reply include/field_specials.h]: I didn't rename that

File: include/party_menu.h (hunk 2 above)
Comment (jiangzhengwenjz) [reply include/party_menu.h]: again

File: src/data/pokemon/cry_ids.h [hunk 4]
```diff
@@ -0,0 +1,138 @@
+static const u16 sSpeciesIdToCryId[] =
+{
+    [SPECIES_TREECKO - 277] = 273,
```
[the remark this answers is not in the document]
Comment (jiangzhengwenjz) [reply src/data/pokemon/cry_ids.h]: It won't match with `HOENN_DEX_{species}`. For example, you may consider `SPECIES_NINJASK - 277 == HOENN_DEX_NINJASK - 1`, but actually that's not true (for many mons). I now call 277 HOENN_MON_SPECIES_START

File: src/data/pokemon/cry_ids.h [hunk 5]
```diff
@@ -0,0 +1,140 @@
+#define HOENN_MON_SPECIES_START 277
+
+static const u16 sHoennSpeciesIdToCryId[] =
+{
+    [SPECIES_TREECKO - HOENN_MON_SPECIES_START] = 273,
```
Comment (PikalaxALT) [inline src/data/pokemon/cry_ids.h]: Perhaps define enums for the Hoenn cry IDs?

Comment (jiangzhengwenjz) [reply src/data/pokemon/cry_ids.h]: Now enums are in include/constants/hoenn_cries.h

File: include/constants/hoenn_cries.h [hunk 6]
```diff
@@ -0,0 +1,143 @@
+#ifndef GUARD_CONSTANTS_HOENN_CRIES_H
+#define GUARD_CONSTANTS_HOENN_CRIES_H
+
+enum
+{
+    CRY_TREECKO = 273,
```
Comment (PikalaxALT) [inline include/constants/hoenn_cries.h]: Sort them in numerical order please

**Pull request merged**